### ISSUE TITLE
Questions - Wasn't sure where else to ask

### ISSUE THREAD

**Issue by GantMan** [mobxjs/mobx-state-tree#575]
First off 👍 on this lib.

Migrating from Redux here, I have some strange questions.

In Redux we had lots of "O NO!" code for when bad data got injected at runtime.   Does Mobx-state-tree have this error handling events for me to tie into?   I've looked through and found efforts to improve error messages etc.  But just wondering if I've completely missed error detection handling, or if this is a path I should document as I go.

Basically middleware error handling.  Is it a thing yet?

**Comment by mattiamanzati** [comment]
"In Redux we had lots of "O NO!" code for when bad data got injected at runtime."
You mean passing a wrong state to the reducer? If so yes, MST will throw and fail if data non applicable to the model is passed :)

"I've completely missed error detection handling, or if this is a path I should document as I go"
I did'nt understood this, can you please provide an example, even in redux, of what you mean? :)

**Comment by GantMan** [comment]
Generally, this problem happens when bad user/API data gets brought into state.

There are libs like: https://github.com/PlatziDev/redux-catch

Are there equivalent systems in mobx-state-tree?

**Comment by mattiamanzati** [comment]
Oh, I see, we have middleware, so it is completely possible, only known plugin at the moment is logrocket which is an official one :)
https://docs.logrocket.com/docs/mobx-state-tree-plugin
We have a middleware logic too, so should be feasible to implement there easily any error logger :)
